TypeCobol/Tools/Options-Config/TypeCobolConfiguration.cs

```diff
@@ -44,29 +140,183 @@ public static OptionSet GetCommonTypeCobolOptions(TypeCobolConfiguration typeCob
+                    config.ExecToStep = ExecutionStep.SemanticCheck;
+                    break;
+                case "4":
+                case "crosscheck":
+                    config.ExecToStep = ExecutionStep.CrossCheck;
+                    break;
+                case "5":
+                case "generate":
+                    config.ExecToStep = ExecutionStep.Generate;
+                    break;
+                default:
+                    errorStack.Add(ReturnCode.ExecToStepError, TypeCobolConfiguration.ErrorMessages[ReturnCode.ExecToStepError]);
+                    break;
+            }
+
+            //// Check required options
+            //InputFileMissing
+            if (config.InputFiles.IsNullOrEmpty())
+                errorStack.Add(ReturnCode.InputFileMissing, TypeCobolConfiguration.ErrorMessages[ReturnCode.InputFileMissing]);
+            //OutputFileMissing/OutputFileError
+            if (config.ExecToStep == ExecutionStep.Generate && !errorStack.ContainsKey(ReturnCode.ExecToStepError))
+            {
+                if (config.OutputFiles.IsNullOrEmpty())
+                    errorStack.Add(ReturnCode.OutputFileMissing, TypeCobolConfiguration.ErrorMessages[ReturnCode.OutputFileMissing]);
+                else if (config.InputFiles.Count != config.OutputFiles.Count)
+                    errorStack.Add(ReturnCode.OutputFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.OutputFileError]);
+            }
+            //SkeletonMissing
+            if (config.ExecToStep == ExecutionStep.Generate && config.skeletonPath.IsNullOrEmpty() && !errorStack.ContainsKey(ReturnCode.ExecToStepError))
+                errorStack.Add(ReturnCode.SkeletonMissing, TypeCobolConfiguration.ErrorMessages[ReturnCode.SkeletonMissing]);
+
+
+            //// Check unexpected token
+            if (unexpectedStrings.Count != 0)
+                errorStack.Add(ReturnCode.UnexpectedParamError, TypeCobolConfiguration.ErrorMessages[ReturnCode.UnexpectedParamError]);
+
+            //// Options values verification
+            //InputFileError
+            foreach (var path in config.InputFiles)
+            {
+                if (FileSystem.GetFiles(path, recursive: false).IsNullOrEmpty() && !errorStack.ContainsKey(ReturnCode.InputFileError))
+                {
+                    errorStack.Add(ReturnCode.InputFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.InputFileError]);
+                }
+            }
+            //SkeletonFileError
+            if (config.ExecToStep == ExecutionStep.Generate && !config.skeletonPath.IsNullOrEmpty() &&  !errorStack.ContainsKey(ReturnCode.ExecToStepError))
+            {
+                if (!File.Exists(config.skeletonPath))
+                {
+                    errorStack.Add(ReturnCode.SkeletonFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.SkeletonFileError]);
+                }
+            }
+            // EncodingError
+            switch (config.RawFormat.ToLower())
```

Comment:
In the previous implementation this option was created using CreateFormat(), why no reusing it now ?